sys/kern/rman.c

```diff
@@ -97,6 +101,13 @@ void rman_fini(rman_t *rm) {
   /* TODO: destroy the `rm_lock` after implementing `mtx_destroy`. */
 }
 
+static int r_share_hit(resource_t *r, rman_addr_t start, size_t count,
+                       res_flags_t flags) {
+  size_t rsize = r->r_end - r->r_start + 1;
+  return r_shareable(r) && (flags & RF_SHAREABLE) && rsize == count &&
+         r->r_start <= start;
+}
```

Comment:
What does 'count' mean? Note that if r_start is strictly smaller than start, than  start+count > r_end, because count == rsize.

Reply:
It's broken. There should be `r->r_start >= start`.  'count' expresses the amount of memory we'd like to allocate.